skyportal/handlers/api/group.py

```diff
@@ -138,22 +141,25 @@ def get(self, group_id=None):
         include_single_user_groups = self.get_query_argument(
             "includeSingleUserGroups", False
         )
-        acls = [acl.id for acl in self.current_user.acls]
         info = {}
         info['user_groups'] = list(self.current_user.groups)
-        info['all_groups'] = (
-            Group.query.all()
-            if "System admin" in acls or "Manage groups" in acls
-            else None
-        )
         if (not include_single_user_groups) or (include_single_user_groups == "false"):
-            info["user_groups"] = [
-                g for g in info["user_groups"] if not g.single_user_group
+            info["all_groups"] = [
+                g
+                for g in self.current_user.accessible_groups
+                if not g.single_user_group
             ]
-            if info["all_groups"]:
-                info["all_groups"] = [
-                    g for g in info["all_groups"] if not g.single_user_group
-                ]
+        else:
+            # `accessible_groups` already includes single user groups for sys admins
+            if "System admin" in self.current_user.permissions:
```

Comment:
If you use set logic to do the joins, as we do in some other place in SkyPortal, there is no need to handle these cases separately.

Reply:
using a single query won't work because accessible_groups doesn't include single user groups for non-admins. or are you referring to something else?

Reply:
if you mean just do the query in either case, then pass through `set` to remove dupicates; yes, I could, but thought it would be more efficient to not have to execute that query unnecessarily in the case of admins

Reply:
OK, this is fine.